packages/dart_frog/lib/src/response.dart

```diff
@@ -61,7 +63,17 @@ class Response {
   Stream<List<int>> bytes() => _response.read();
 
   /// Returns a [Future] containing the body as a [String].
-  Future<String> body() => _response.readAsString();
+  Future<String> body() async {
+    if (_bodyCompleter == null) {
+      _bodyCompleter = Completer<String>();
+      try {
+        _bodyCompleter!.complete(await _response.readAsString());
```

Comment:
hum I wonder if this actually works, because here you read it anyway, so shelf can't read it too to send back the response right? So not sure that's the way to go for this

Reply:
My go to for response would be to not create the shelf response until shelf need it to answer the request. but no idea if that would work that's just an idea

Reply:
added some more tests around this, let me know what you think 👍

Reply:
Now that the response is cloned I guess it should work, I have no time to test it (maybe this weekend) but to me should be good, if it doesn't I'll open a new issue, so this don't stay open to long. Thanks for this!